test/Feature/HLSLLib/distance.32.denormal.test

```diff
@@ -0,0 +1,91 @@
+#--- source.hlsl
+
+StructuredBuffer<float4> X : register(t0);
+StructuredBuffer<float4> Y : register(t1);
+
+RWStructuredBuffer<float> Result : register(u2);
+
+[numthreads(1,1,1)]
+void main() {
+  // test special cases
+  // distance({inf, 0.0}, {0.0, 1.0}) = inf
+  Result[0] = distance(X[0].xw, Y[0].wx);
+
+  // distance({NaN, 0.0}, {1.0, 0.0}) = NaN
+  Result[1] = distance(X[0].yw, Y[0].xw);
+
+  // distance({-inf, 0.0}, {0.0, 1.0}) = inf
+  Result[2] = distance(X[0].zw, Y[0].wx);
+
+  // distance({inf, 0.0}, {0.0, NaN}) = NaN
+  Result[3] = distance(X[0].xw, X[0].wy);
+
+  // distance({inf, NaN, -inf, 0}, {1.0, -denorm, denorm, 0}) = NaN
+  Result[4] = distance(X[0].xyzw, Y[0].xyzw);
+
+  // distance({denorm, denorm}, {1.0, 0.0}) = 1.0
+  Result[5] = distance(Y[0].zz, Y[0].xw);
+
+  // distance({-denorm, -denorm}, {0.0, 1.0}) = 1.0
+  Result[6] = distance(Y[0].yy, Y[0].wx);
+}
+
+//--- pipeline.yaml
+
+---
+Shaders:
+  - Stage: Compute
+    Entry: main
+    DispatchSize: [1, 1, 1]
+Buffers:
+  - Name: X
+    Format: Float32
+    Stride: 16
+    Data: [ inf, NaN, -inf, 0 ]

```

Comment:
Would -0 also an interesting value to test?

Reply:
It would, looks like it didn't affect any of the expected results however.